components/tabs/src/sync/bridge.rs

```diff
@@ -192,65 +218,86 @@ impl TabsBridgedEngine {
         Self { bridge_impl }
     }
 
-    pub fn last_sync(&self) -> Result<i64> {
+    pub fn last_sync(&self) -> ApiResult<i64> {
         self.bridge_impl.last_sync()
     }
 
-    pub fn set_last_sync(&self, last_sync: i64) -> Result<()> {
+    pub fn set_last_sync(&self, last_sync: i64) -> ApiResult<()> {
         self.bridge_impl.set_last_sync(last_sync)
     }
 
-    pub fn sync_id(&self) -> Result<Option<String>> {
+    pub fn sync_id(&self) -> ApiResult<Option<String>> {
         self.bridge_impl.sync_id()
     }
 
-    pub fn reset_sync_id(&self) -> Result<String> {
+    pub fn reset_sync_id(&self) -> ApiResult<String> {
         self.bridge_impl.reset_sync_id()
     }
 
-    pub fn ensure_current_sync_id(&self, sync_id: &str) -> Result<String> {
+    pub fn ensure_current_sync_id(&self, sync_id: &str) -> ApiResult<String> {
         self.bridge_impl.ensure_current_sync_id(sync_id)
     }
 
-    pub fn prepare_for_sync(&self, client_data: &str) -> Result<()> {
+    pub fn prepare_for_sync(&self, client_data: &str) -> ApiResult<()> {
         self.bridge_impl.prepare_for_sync(client_data)
     }
 
-    pub fn sync_started(&self) -> Result<()> {
+    pub fn sync_started(&self) -> ApiResult<()> {
         self.bridge_impl.sync_started()
     }
 
-    pub fn store_incoming(&self, incoming: Vec<String>) -> Result<()> {
-        let mut envelopes = Vec::with_capacity(incoming.len());
-        for inc in incoming {
-            envelopes.push(serde_json::from_str::<IncomingEnvelope>(&inc)?);
+    // Decode the JSON-encoded incoming envelopes that UniFFI passes to us
+    fn convert_incoming_envelopes(
+        &self,
+        incoming: Vec<String>,
+    ) -> ApiResult<Vec<IncomingEnvelope>> {
+        handle_error! {
+            let mut envelopes = Vec::with_capacity(incoming.len());
+            for inc in incoming {
+                envelopes.push(serde_json::from_str::<IncomingEnvelope>(&inc)?);
+            }
+            Ok(envelopes)
         }
```

Comment:
This is a papercut from using the `handle_error!` macro.  It would be simpler if this code was in `store_incoming()` since that's the only place it's used.  However, I couldn't get the `handle_error!` macro to work there because it needs to wrap the body of the entire function, but that didn't work because `self.bridge_impl.store_incoming()` is already returning an `ApiResult`.

It's a bit annoying, but the fact that this is the biggest issue that I've seen with that macro seems to indicate it's working out well.